### PR TITLE
Lucas user hours set to zero

### PR DESCRIPTION
**Pull request by Lucas-E**
This PR solves (PRIORITY LOW) Jae: Special Situation - User Hours Set to Zero

### PLEASE USE THE PROPER BACKEND ([#298](https://github.com/OneCommunityGlobal/HGNRest/pull/298)) SO THE PR CAN WORK PROPERLY

<hr>

### Main changes

Now, Mentor users and users with zero hours are not displayed on the leaderboard and do not add to the HGN totals. Mentor users can only be seen by themselves and Owners, Administrators and Core Teams on the Leaderboard. Also, Mentor users and users with zero hours can see the members on their respective teams.

<hr>

### How to test

- Checkout the current branch on both frontend and backend provided on Slack and on this PR
- npm install (on the backend)
- npm run build (on the backend)
- npm run start (on the backend)
- npm install (on the frontend)
- npm start:local (on the frontend)
- Login as various roles and check the users that appear on the Leaderboard (Mentors and zero hour users should see themselves and team members, Owners, Administrators and Core Team members should be able to see all users, including zero hour users and Mentor users. Other role's users should not see Mentor users and zero hour users)
- Change some user's hours to zero and role to Mentor and check if the the count on HGN Totals changes.

### CODE REVIEW

**Review by nicollext — COMMENTED** [review]
Tested all cases and it is working as expected. 

I just have one doubt, about this case:
    //if the user is not an administrator, nor owner, nor mentor, and also not a core team, it will only see the people from the same team, no zero hour members and not mentor members

When logged as a Volunteer or Manager I can see the Mentor that is at my team. Is this the expected? 
lucas_user_hours_set_to_zero_Volunteer2
lucas_user_hours_set_to_zero_Manager

For other roles it is working as expected. 

1. Administrator, Owner and Core Team are  able to see all users, including zero hour users and Mentor users.

https://user-images.githubusercontent.com/101591708/226692519-385ac921-38cf-45b0-8c70-1a950104cad8.MP4

https://user-images.githubusercontent.com/101591708/226692594-fb93dcee-3d6f-41d2-adcc-ff1387e4adfb.MP4

https://user-images.githubusercontent.com/101591708/226692583-231eedc0-6bb9-46cd-a1be-ac24c7d29d29.MP4

2. Mentors and zero hour users can see themselves and team members. Other role's users should not see Mentor users and zero hour users.

lucas_user_hours_set_to_zero_Mentor

lucas_user_hours_set_to_zero_Volunteer2

3. Mentor users and users with zero hours do not add to the HGN totals.

https://user-images.githubusercontent.com/101591708/226696799-61ac7e20-b6e3-4883-a879-c58f786a2e9c.MP4

**Comment by Lucas-E** [comment]
> Tested all cases and it is working as expected.
> 
> I just have one doubt, about this case: //if the user is not an administrator, nor owner, nor mentor, and also not a core team, it will only see the people from the same team, no zero hour members and not mentor members
> 
> When logged as a Volunteer or Manager I can see the Mentor that is at my team. Is this the expected? lucas_user_hours_set_to_zero_Volunteer2 lucas_user_hours_set_to_zero_Manager
> 
> For other roles it is working as expected.
> 
> 1. Administrator, Owner and Core Team are  able to see all users, including zero hour users and Mentor users.
> 
> 2. Mentors and zero hour users can see themselves and team members. Other role's users should not see Mentor users and zero hour users.
> 
> lucas_user_hours_set_to_zero_Mentor
> 
> lucas_user_hours_set_to_zero_Volunteer2
> 
> 3. Mentor users and users with zero hours do not add to the HGN totals.

Hey Nicolle, thanks for the review! Could you please dm me a console print with the expanded array in it on slack?

**Comment by nicollext** [comment]
> > Tested all cases and it is working as expected.
> > I just have one doubt, about this case: //if the user is not an administrator, nor owner, nor mentor, and also not a core team, it will only see the people from the same team, no zero hour members and not mentor members
> > When logged as a Volunteer or Manager I can see the Mentor that is at my team. Is this the expected? lucas_user_hours_set_to_zero_Volunteer2 lucas_user_hours_set_to_zero_Manager
> > For other roles it is working as expected.
> > 
> > 1. Administrator, Owner and Core Team are  able to see all users, including zero hour users and Mentor users.
> > 2. Mentors and zero hour users can see themselves and team members. Other role's users should not see Mentor users and zero hour users.
> > 
> > lucas_user_hours_set_to_zero_Mentor
> > lucas_user_hours_set_to_zero_Volunteer2
> > 
> > 3. Mentor users and users with zero hours do not add to the HGN totals.
> 
> Hey Nicolle, thanks for the review! Could you please dm me a console print with the expanded array in it on slack?

Of course ;)

**Review by nicollext — APPROVED** [review]

**Review by kangjay88 — DISMISSED** [review]
Hi @Lucas-E , I have tested this PR with all roles except 'Owner', and it is working as intended. Nice code implementation, here are my examples:
Volunteer: 
PR727-1
Manager:
PR727-2
CoreTeam:
https://user-images.githubusercontent.com/108297706/227730855-42046f81-b891-423d-9b88-9ac5c0d71305.mp4
Admin:
https://user-images.githubusercontent.com/108297706/227730923-5b4df2a8-4939-433a-89e3-666ee17559c6.mp4
Mentor:
PR727-5

**Review by vitorsaa2k — COMMENTED** [review]
Tested with Admin, Core Team, Owner and Volunteer, all cases works as intended, user with the Mentor role can not be seem by volunteer users but the other roles can. Mentor can not be seem by another Mentor. Zero hour users can be seem by Admin or Owners user, but the other roles can't.

**Comment by Lucas-E** [comment]
@one-community

**Comment by one-community** [comment]
Excellent, thank you everyone. Moving this pair to final review.

**Review by rauleffting — CHANGES_REQUESTED** [review]
Hi, Lucas. Your PR is perfect and worked as expected. However, you still need to delete a debug output that was left. 

https://user-images.githubusercontent.com/29555732/230135008-0192ae98-917a-43b3-9e36-a695282778ff.mp4

Thank you very much!

**Review by Elisaudo — APPROVED** [review]
Hi, Lucas, I tested all cases and it is working as expected.

1)Owners, Administrators and Core Team members should be able to see all users, including zero hour users and Mentor users.

[PR#727 PR#298 Video 1.webm](https://user-images.githubusercontent.com/64317219/231634894-914599fd-99cc-4e4d-9b8c-51306883a4ad.webm)

2)	Mentor users can only be seen by themselves and Owners, Administrators and Core Teams on the Leaderboard. 

PR#727 PR#298 - Mentor users can only be seen by themselves

PR#727 PR#298 - Mentor users can only be seen by themselves 2

3)	Mentor users and users with zero hours are not displayed on the leaderboard and do not add to the HGN totals. Other role's users should not see Mentor users and zero hour users. (manager, volunteer)

PR#727 PR#298 - Other role's cannot see Mentor users and zero hour users

4)	Mentor users and users with zero hours can see the members on their respective teams.

PR#727 PR#298 - Mentor users and users with zero hours can see team members 1



PR#727 PR#298 - Mentor users and users with zero hours can see team members 2



PR#727 PR#298 - Elisaudo Team - team members

File: src/components/LeaderBoard/LeaderBoard.container.jsx [hunk 1]
```diff
@@ -6,6 +6,41 @@ import { get, round, maxBy } from 'lodash';
 
 const mapStateToProps = state => {
   let leaderBoardData = get(state, 'leaderBoardData', []);
+  let user = get(state, 'userProfile', []);
+  console.log(leaderBoardData)
```
[the remark this answers is not in the document]
Comment (Lucas-E) [reply src/components/LeaderBoard/LeaderBoard.container.jsx]: Done

**Review by rauleffting — APPROVED** [review]
Hi, @Lucas-E. It worked perfectly.

pic

Approved!

**Comment by one-community** [comment]
Excellent job on this!

**Pull request merged**